### PR TITLE
feat: Qdrant Integration

### LINKED ISSUE #7670: [Integration] Integrate Qdrant (vector db) with MindsDB
The goal of this issue is to build a new handler that will connect MindsDB to Qdrant vector database.

## Resources :bulb: 

:checkered_flag: To get started, please refer to the following resources:

* :books:  [Building Data Handler docs](https://docs.mindsdb.com/contribute/data-handlers): This documentation will guide you through the process of building a custom handler for MindsDB. It contains essential information on the handler structure, methods, and best practices.

* :book:  [Qdrant](https://qdrant.tech/)

* :point_right: [ChromaDB](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/chromadb_handler): This existing handler connects MindsDB with ChromaDB.

## Next Steps :male_detective: :female_detective: 
 
-   You can familiarize yourself with Qdrant.
-   Create a new handler that connects MindsDB to Qdrant.
-   Implement the core features mentioned above.
-   Thoroughly test the integration to ensure proper functionality.
-   Submit a pull request with the new Qdrant Integration handler.

## :loudspeaker: Additional rewards :trophy:

Each App integration brings :two: :five:  : points for SWAG and entry into the draw for a  :point_down: :
-  :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### PR DESCRIPTION
**Pull request by Anush008**
## Description
This PR intends to add Qdrant as a supported vector database to MindsDB.

Resolves #7670.

## Type of change
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [x] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

* You can set the "location" value when connecting to Qdrant as ":memory:", which will instantiate a in-memory instance, making it easier for you to try it out.
* As mentioned in https://github.com/mindsdb/mindsdb/issues/7670#issuecomment-1761198690, collection creation config has to be passed when connecting to an instance.
* Usage docs can be found in the [README.md](https://github.com/Anush008/mindsdb/blob/qdrant-minds-int/mindsdb/integrations/handlers/qdrant_handler/README.md).

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

cc @dusvyat.

### CODE REVIEW

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by Anush008** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by dusvyat** [comment]
@Anush008 - looks good, unit tests are passing on my end.

A minor non-blocking point, any issue with creating an issue for 'adding support for update' and 'add support for payload indexing'. That way we can make sure it isn't forgotten about

Thanks!

File: mindsdb/integrations/handlers/qdrant_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+WHERE `metadata.source` = 'bbc';
+```
+
+#### Drop a table
+```sql
+ DROP TABLE qdrant_test.test_table;
+```
+
+## NOTICE
+Qdrant supports payload indexing that vastly improves retrieval efficiency with filters and is highly recommended. Please note that this feature currently cannot be configured via MindsDB and must be set up separately if needed.
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/qdrant_handler/README.md:81]: please create an issue so we can not forget about supporting this in future

File: mindsdb/integrations/handlers/qdrant_handler/qdrant_handler.py [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+            )
+
+        return models.Filter(must=qdrant_filters) if qdrant_filters else None
+
+    def update(
+        self, table_name: str, data: pd.DataFrame, columns: List[str] = None
+    ) -> HandlerResponse:
+        """
+        Update data in the Qdrant database.
+        TODO: Update for vector DBs has not been implemented.
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/qdrant_handler/qdrant_handler.py:283]: please create an issue so we can not forget about supporting this in future

**Review by dusvyat — APPROVED** [review]

File: tests/handler_tests/test_qdrant_handler.py [hunk 3]
```diff
@@ -0,0 +1,445 @@
+from unittest.mock import patch
+
+import pytest
+import pandas as pd
+from mindsdb_sql import parse_sql
+
+from unit.executor_test_base import BaseExecutorTest
```
Comment (dusvyat) [inline tests/handler_tests/test_qdrant_handler.py:7]: should be` from unit.executor_test_base import BaseExecutorTest` as tests are executed from package root

File: mindsdb/integrations/handlers/qdrant_handler/README.md (hunk 1 above)
Comment (Anush008) [reply mindsdb/integrations/handlers/qdrant_handler/README.md:81]: #7863.

File: mindsdb/integrations/handlers/qdrant_handler/qdrant_handler.py (hunk 2 above)
Comment (Anush008) [reply mindsdb/integrations/handlers/qdrant_handler/qdrant_handler.py:283]: #7864.

File: tests/handler_tests/test_qdrant_handler.py (hunk 3 above)
Comment (Anush008) [reply tests/handler_tests/test_qdrant_handler.py]: https://github.com/mindsdb/mindsdb/pull/7833/commits/183214419563b7259b8c8c263d5cd4e7c535d3b3

**Pull request merged**